connector/connect/server/src/main/scala/org/apache/spark/sql/connect/ui/SparkConnectServerListener.scala

```diff
@@ -128,6 +126,43 @@ private[connect] class SparkConnectServerListener(
     }
   }
 
+  def onSQLExecutionStart(e: SparkListenerSQLExecutionStart): Unit = {
+    val executeJobTagOpt = e.jobTags.find {
+      case ExecuteJobTag(_) => true
+      case _ => false
+    }
+    if (executeJobTagOpt.isEmpty) {
+      return
+    }
+    val executeJobTag = executeJobTagOpt.get
+    val exec = executionList.get(executeJobTag)
+    if (exec.nonEmpty) {
+      exec.foreach { exec =>
+        exec.sqlExecId += e.executionId.toString
+        updateLiveStore(exec)
+      }
+    } else {
+      // This block guards against potential event re-ordering.
```

Comment:
Can you explain the event re-ordering here?

Reply:
It's to guard against SQLStart and JobStart events potentially coming after an OperationClosed event. It's mirrors the one in onJobStart (which was included in the Thrift Server listener as well). See the comment here: https://github.com/apache/spark/pull/42244/files#diff-ffa4c6fbd382a52c08a0f10b7fbbd6c2d09de322f7e3145ecba464f0cda5843dR89-R92

Reply:
Let's extend the comment in the code

Reply:
BTW, how often can the event reorder happen? It looks odd to update the data in KVStore.